test-framework/junit5-component/src/main/java/io/quarkus/test/component/QuarkusComponentTestExtension.java

```diff
@@ -1051,21 +1057,26 @@ static <T> T cast(Object obj) {
 
     private List<FieldInjector> injectFields(Class<?> testClass, Object testInstance) throws Exception {
         List<FieldInjector> injectedFields = new ArrayList<>();
-        for (Field field : findInjectFields(testClass)) {
+        for (Field field : findInjectFields(testClass, false)) {
             injectedFields.add(new FieldInjector(field, testInstance));
         }
         return injectedFields;
     }
 
-    private List<Field> findInjectFields(Class<?> testClass) {
+    private List<Field> findInjectFields(Class<?> testClass, boolean scanEnclosingClasses) {
         List<Class<? extends Annotation>> injectAnnotations;
         Class<? extends Annotation> deprecatedInjectMock = loadDeprecatedInjectMock();
         if (deprecatedInjectMock != null) {
             injectAnnotations = List.of(Inject.class, InjectMock.class, deprecatedInjectMock);
         } else {
             injectAnnotations = List.of(Inject.class, InjectMock.class);
         }
-        return findFields(testClass, injectAnnotations);
+        List<Field> found = findFields(testClass, injectAnnotations);
+        if (scanEnclosingClasses && testClass.getEnclosingClass() != null) {
+            // @Nested test class
+            found.addAll(findFields(testClass.getEnclosingClass(), injectAnnotations));
```

Comment:
You should loop until there's no enclosing class to support multiple nesting.

Reply:
Eh, you're right. I thought that it scans the classes recursively :facepalm:. Anyway, I will add a nested level 2 test ;-).

Reply:
Done.